OrbitGl/App.cpp

```diff
@@ -817,36 +779,77 @@ void OrbitApp::LoadModules(const std::shared_ptr<Process>& process, int32_t proc
     // TODO (159889010) Move symbol loading off the main thread.
     const std::string& module_path = module->m_FullName;
     const std::string& build_id = module->m_DebugSignature;
-    if (build_id.empty()) {
-      LOG("Warning: Module \"%s\" does not contain a build id. Orbit cannot use local symbol files "
-          "or cache symbols without a build id.",
-          module_path);
-      LoadModuleOnRemote(process, process_id, module, preset);
-      continue;
-    }
 
-    auto scoped_status = CreateScopedStatus(
-        absl::StrFormat("Trying to find symbols for \"%s\" on local machine...", module_path));
-    auto symbols = symbol_helper_.LoadUsingSymbolsPathFile(module_path, build_id);
+    // If module has a build id, local loading can be attempted;
+    if (!build_id.empty()) {
+      const auto scoped_status = CreateScopedStatus(absl::StrFormat(
+          "Searching for symbols on local machine (module: \"%s\")...", module_path));
 
-    // Try loading from the cache
-    if (!symbols) {
-      const std::string cached_file_name = symbol_helper_.GenerateCachedFileName(module_path);
-      symbols = symbol_helper_.LoadSymbolsFromFile(cached_file_name, build_id);
+      {
+        // Search in user provided symbol directories (via SymbolsPaths.txt from config folder)
+        const auto symbols_path =
+            symbol_helper_.FindSymbolsWithSymbolsPathFile(module_path, build_id);
+        if (symbols_path) {
+          LOG("Found symbols for module \"%s\" in user provided symbol folder. symbols filename: "
+              "\"%s\"",
+              module_path, symbols_path.value());
+          LoadSymbols(symbols_path.value(), process, module, preset);
+          continue;
+        }
+      }
+      {
+        // Search in cache
+        const auto symbols_path = symbol_helper_.FindSymbolsInCache(module_path, build_id);
+        if (symbols_path) {
+          LOG("Found symbols for module \"%s\" in cache. symbols filename: \"%s\"", module_path,
+              symbols_path.value());
+          LoadSymbols(symbols_path.value(), process, module, preset);
+          continue;
+        }
+      }
```

Comment:
maybe extract the part inside this to a separate method (too much nesting)?

Reply:
or remove the additional scope and reassign symbols_path

Reply:
I extracted the finding of files into a `FindSymbolsLocally` method.